i18n/scripts/upload.js

```diff
@@ -0,0 +1,64 @@
+const fs = require('fs');
+const path = require('path');
+const request = require('request-promise');
+
+// Get files to upload from API
+const files = require('./config').files;
+if (!files[0]) {
+  console.log('⚠️  missing file to upload, aborting.');
+  process.exit(1);
+}
+
+// Get project ID & key from environment variables
+const { CROWDIN_API_KEY, CROWDIN_PROJECT_IDENTIFIER } = process.env;
+if (!CROWDIN_API_KEY || !CROWDIN_PROJECT_IDENTIFIER) {
+  console.log(
+    '⚠️  missing necessary environment variables (CROWDIN_API_KEY, CROWDIN_PROJECT_IDENTIFIER), aborting.',
+  );
+  process.exit(1);
+}
+
+// Run the uploader for each file we got from config
+for (file of files) {
+  createOrUpdateFile(...file);
+}
+
+// Attempt to update the file (most common operation), default to creating it
+// if it does not exist (initial setup)
+function createOrUpdateFile(fileName, filePath, shouldCreateFile) {
+  console.log(
+    `🙈 uploading ${fileName} [${shouldCreateFile ? 'CREATE' : 'UPDATE'}]...`,
+  );
+
+  // Crowdin expects the POST request to contain the files in form data format
+  const formData = {
+    [`files[${fileName}]`]: fs.createReadStream(
+      path.join(__dirname + filePath),
+    ),
+    json: JSON.stringify(true),
+  };
+
+  // Crowdin API action keywords/routes
+  const keyword = shouldCreateFile ? 'add-file' : 'update-file';
+
+  request
+    .post({
+      formData,
+      url: `https://api.crowdin.com/api/project/${CROWDIN_PROJECT_IDENTIFIER}/${keyword}?key=${CROWDIN_API_KEY}`,
+    })
+    .then(
+      () => console.log(`🚀 uploaded ${fileName}.`),
+      response => {
+        const error = JSON.parse(response.error);
+        if (error.error.code === 8) {
+          console.log(
+            `🙉 cannot update ${fileName}, attempting to create instead...`,
+          );
+          return createOrUpdateFile(fileName, filePath, true);
+        }
+
+        console.log(`💥 failed to upload ${fileName}, aborting.`);
+        process.exit(1);
+      },
+    );
+}
```

Comment:
Is the default exit code 0 for a node script?

Reply:
It is